src/components/controls/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,119 @@
+'use client';
+
+import React from 'react';
+
+import {Eye, EyeSlash} from '@gravity-ui/icons';
+
+import {ActionTooltip} from '../../ActionTooltip';
+import {Button} from '../../Button';
+import {ClipboardButton} from '../../ClipboardButton';
+import {Icon} from '../../Icon';
+import {block} from '../../utils/cn';
+import {TextInput} from '../TextInput';
+import type {TextInputProps} from '../TextInput';
+
+import {i18n} from './i18n';
+import {getActionButtonSizeAndIconSize} from './utils';
+
+import './PasswordInput.scss';
+
+const b = block('password-input');
+
+export type PasswordInputProps = Omit<TextInputProps, 'type'> & {
+    /** Show copy button */
+    showCopyButton?: boolean;
+    /** Show reveal button */
+    showRevealButton?: boolean;
+    /** Disable the tooltip for the copy button. The tooltip will not be displayed */
+    hasCopyTooltip?: boolean;
+    /** Disable the tooltip for the reveal button. The tooltip will not be displayed */
+    hasRevealTooltip?: boolean;
+};
+
+export const PasswordInput = (props: PasswordInputProps) => {
```

Comment:
Now uncontrolled component does not work as expected. Let's add

```js
const [value, setValue] = useControlledState(props.value, props.defaultValue, props.onUpdate);
```

to be able to control the state of the value.